api/internal/postgres/postgres.go

```diff
@@ -0,0 +1,359 @@
+package postgres
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+
+	"github.com/equinor/oneseismic/api/internal"
+)
+
+type coordinate struct {
+	X float64
+	Y float64
+}
+
+type Point coordinate
+
+type Linestring struct {
+	Coords []coordinate
+}
+
+type Polygon struct {
+	Coords []coordinate
+}
+
+type Geometry struct {
+	Point      *Point
+	Linestring *Linestring
+	Polygon    *Polygon
+}
+
+type Manifest struct {
+	FormatVersion  *int32        `json:"format-version,omitempty"`
+	UploadFilename *string       `json:"upload-filename,omitempty"`
+	Guid           *string       `json:"guid,omitempty"`
+	Data           *[]Data       `json:"data,omitempty"`
+	Attributes     *[]Attribute  `json:"attributes,omitempty"`
+	LineNumbers    *[][]int32    `json:"line-numbers,omitempty"`
+	LineLabels     *[]string     `json:"line-labels,omitempty"`
+	SampleValueMin *float64      `json:"sample-value-min,omitempty"`
+	SampleValueMax *float64      `json:"sample-value-max,omitempty"`
+}
+
+type Data struct {
+	FileExtension *string    `json:"file-extention,omitempty"`
+	Filters       *[]string  `json:"filters,omitempty"`
+	Shapes        *[][]int32 `json:"shapes,omitempty"`
+	Prefix        *string    `json:"prefix,omitempty"`
+	Resolution    *string    `json:"resolution,omitempty"`
+}
+
+type Attribute struct {
+	Type          *string    `json:"type,omitempty"`
+	Layout        *string    `json:"layout,omitempty"`
+	FileExtension *string    `json:"file-extention,omitempty"`
+	Labels        *[]string  `json:"labels,omitempty"`
+	Shapes        *[][]int32 `json:"shapes,omitempty"`
+	Prefix        *string    `json:"prefix,omitempty"`
+}
+
+type ManifestFilter struct {
+	Eq  *Manifest
+	Or  *[]ManifestFilter
+	And *[]ManifestFilter
+}
+
+type operator string
+
+const (
+	And operator = " AND "
+	Or  operator = " OR "
+)
+
+
+/*
+ * Stupid helper to join together terms by some operator
+ */
+func join(op operator, terms ...*string) (*string) {
+	// Remove all nil values before combining
+	_terms := make([]string, 0)
+	for _, elem := range terms {
+		if elem != nil {
+			_terms = append(_terms, *elem)
+		}
+	}
+
+	if len(_terms) == 0 {
+		return nil
+	}
+
+	out := strings.Join(_terms, string(op))
+	return &out
+}
+
+/*
+ *  Map a manifest-instance (filter) to SQL using the jsonb containment
+ *  operator ( @> )
+ */
+func containsFilter(filter *Manifest, dbColumn string) (*string) {
+	if filter == nil {
+		return nil
+	}
+
+	b, _ := json.Marshal(filter)
```

Comment:
The only way this can fail is if:
> Channel, complex, and function values cannot be encoded in JSON. Attempting to encode such a value causes Marshal to return an UnsupportedTypeError.
JSON cannot represent cyclic data structures and Marshal does not handle them. Passing cyclic structures to Marshal will result in an error.

So i.e. it can only fail when you have data structures that are not supported by JSON. And since we Marshall the manifest type, which is modelled after the manifest.json it's impossible for it to fail. Now, or in the future. Unless we make an illegal `manifest.json` that is.